handler/dreamhost/dreamhost_handler.go

```diff
@@ -0,0 +1,124 @@
+package dreamhost
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"runtime/debug"
+	"strings"
+	"time"
+
+	"github.com/TimothyYe/godns"
+	"github.com/google/uuid"
+)
+
+var (
+	// DreamhostURL the API address for dreamhost.com
+	DreamhostURL = "https://api.dreamhost.com"
+)
+
+// Handler struct
+type Handler struct {
+	Configuration *godns.Settings
+}
+
+// SetConfiguration pass dns settings and store it to handler instance
+func (handler *Handler) SetConfiguration(conf *godns.Settings) {
+	handler.Configuration = conf
+}
+
+// DomainLoop the main logic loop
+func (handler *Handler) DomainLoop(domain *godns.Domain, panicChan chan<- godns.Domain) {
+	defer func() {
+		if err := recover(); err != nil {
+			log.Printf("Recovered in %v: %v\n", err, debug.Stack())
+			panicChan <- *domain
+		}
+	}()
+
+	for {
+		currentIP, err := godns.GetCurrentIP(handler.Configuration)
+
+		if err != nil {
+			log.Println("get_currentIP:", err)
+			continue
+		}
+		log.Println("currentIP is:", currentIP)
+
+		for _, subDomain := range domain.SubDomains {
+			hostname := subDomain + "." + domain.DomainName
+			lastIP := godns.ResolveDNS(hostname, handler.Configuration.Resolver)
+			//check against currently known IP, if no change, skip update
+			if currentIP == lastIP {
+				log.Printf("IP is the same as cached one. Skip update.\n")
+			} else {
+				log.Printf("%s.%s Start to update record IP...\n", subDomain, domain.DomainName)
+				handler.UpdateIP(hostname, currentIP, lastIP)
+
+				// Send notification
+				if err := godns.SendNotify(handler.Configuration, fmt.Sprintf("%s.%s", subDomain, domain.DomainName), currentIP); err != nil {
+					log.Println("Failed to send notification")
+				}
+			}
+		}
+
+		// Sleep with interval
+		log.Printf("Going to sleep, will start next checking in %d seconds...\r\n", handler.Configuration.Interval)
+		time.Sleep(time.Second * time.Duration(handler.Configuration.Interval))
+	}
+
+}
+
+// UpdateIP update subdomain with current IP
+func (handler *Handler) UpdateIP(hostname, currentIP, lastIP string) {
+
+	handler.updateDNS(lastIP, currentIP, hostname, "remove")
+	handler.updateDNS(lastIP, currentIP, hostname, "add")
+
+}
+
+// updateDNS can add or remove DNS records.
+func (handler *Handler) updateDNS(dns, ip, hostname, action string) {
+	// Generates UUID
+	uid, _ := uuid.NewRandom()
+	values := url.Values{}
+	values.Add("record", hostname)
+	values.Add("key", handler.Configuration.LoginToken)
+	values.Add("type", "A")
```

Comment:
Does Dreamhost DNS support IPV6? 
It means that we can update domain record with type `AAAA` here.

Reply:
Good catch! Done.